fenixedu-ist-integration/src/main/java/pt/ist/fenixedu/integration/domain/BpiCard.java

```diff
@@ -0,0 +1,43 @@
+package pt.ist.fenixedu.integration.domain;
+
+import java.time.Year;
+
+import org.fenixedu.bennu.core.domain.User;
+import org.fenixedu.bennu.core.security.Authenticate;
+import org.joda.time.DateTime;
+
+import pt.ist.fenixframework.Atomic;
+
+public class BpiCard extends BpiCard_Base {
+
+    public BpiCard(User user, boolean allow) {
+        super();
+        setUser(user);
+        setAllowSendDetails(allow);
+    }
+
+    @Atomic
+    public static void setGrantAccess(final boolean allowAccess) {
+        final User user = Authenticate.getUser();
```

Comment:
The user should be a parameter, and when calling from controller pass it through.